src/cljs/nnts2/routes.cljs

```diff
@@ -16,12 +16,14 @@
     (re-frame/dispatch [::events/set-active-panel :home-panel])
     (re-frame/dispatch [::events/get-user-info])))
 
-(defroute "/note" []
+(defroute "/notes" []
   (do
-    (re-frame/dispatch [::events/set-active-panel :note-panel])))
+    (re-frame/dispatch [::events/set-active-panel :note-panel])
```

Comment:
Dispatch a single event that indicates that `/notes` was navigated to, and return effects to set the active panel and get the notes in the handler of this event. This decouples the routing definition from the navigation behaviour.

Decoupling the route definition from the navigation behaviour makes the navigation behaviour trivial to test.

Ideally we'd want to configure our routing library to automatically dispatch an event with an appropriate parameter whenever any route is navigated to. I'm not sure how to do this though.

Reply:
Now, there is no notes route separately. So instead of having an event that 'notes was navigated to', I am calling an event 'note panel loaded' after the panel is loaded (as part of the home function).
This event in turn triggers the 'get notes list' event and any other events you might want to associate with this  panel.